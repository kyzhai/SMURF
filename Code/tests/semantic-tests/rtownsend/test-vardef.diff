3c3
< 		ID: a :: Num
---
> 		ID: a :: Int
11c11
< 	ID: a :: Num
---
> 	ID: a :: Int
