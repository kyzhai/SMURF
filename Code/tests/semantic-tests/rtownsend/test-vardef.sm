/* Variable definitions don't require type signatures */

a = 3

b = a

main = []

