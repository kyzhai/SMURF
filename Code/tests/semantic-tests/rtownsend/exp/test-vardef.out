Program: 
	SVardef: 
		ID: a  :: Int

	3
	SVardef: 
		ID: b  :: Int

	a
	SMain: []
Global Scope: 
	ID: a  :: Int
	ID: b  :: Int
	ID: main  :: Unknown
	ID: print x :: a -> a
	ID: random  :: Int

